managers/s3.py

```diff
@@ -37,11 +42,79 @@ def createS3Bucket(self, bucketName, bucketPermissions):
         raise S3Error('Unable to create bucket in s3')
 
 
-    def createManifestInS3(self, manifestPath, manifestJSON, s3_bucket: str):
+    def store_pdf_manifest(self, record, bucket_name, requires_login: bool):
+        for link in record.has_part:
+            item_no, uri, source, media_type, _ = link.split('|')
+
+            if media_type == 'application/pdf':
+                record_id = record.identifiers[0].split('|')[0]
+
+                if requires_login:
+                    manifest_path = f'manifests/publisher_backlist/{source}/{record_id}.json'
+                else:
+                    manifest_path = f'manifests/{source}/{record_id}.json''manifests/{}/{}.json'
+
+                manifest_uri = get_stored_file_url(storage_name=bucket_name, file_path=manifest_path)
+
+                manifest_json = self.generate_manifest(record, uri, manifest_uri)
+
+                self.create_manifest_in_s3(manifest_path, manifest_json, bucket_name)
+
+                if source == 'clacso':
+                    link_string = Part(
+                        index=item_no, 
+                        url=manifest_uri,
+                        source=source,
+                        file_type='application/webpub+json',
+                        flags=json.dumps(dataclasses.asdict(FileFlags()))
+                    ).to_string()
```

Comment:
I think we should try to not have any source specific logic in this function.

Reply:
What is your suggestion to keep CLACSO books from being available on the web reader if we don't want source specific logic with the flags all being false in this function?

Reply:
Also for CLACSO, if it's not readable, do we need the manifest?

Reply:
I'll pass the flags as a parameter in my next commit.  Also, we don't need the manifest for non-readble books from talking to Jackie since they should only be used for the web reader and the download comes from item links found by the item download flag. We will need these manifests when we decide to add the read online button back for the CLACSO books so I suggest we keep them and then update the item download flags for them. What are your thoughts?

Reply:
That makes sense - we can always reingest these records, too.